engine/internal/modeldownloader/huggingface/download.go

```diff
@@ -8,18 +8,40 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"strings"
 
+	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"github.com/aws/smithy-go"
 )
 
 const siFilename = "model.safetensors.index.json"
 
 type s3Client interface {
 	Download(ctx context.Context, f io.WriterAt, path string) error
+	ListObjectsPages(ctx context.Context, prefix string, f func(page *s3.ListObjectsV2Output, lastPage bool) bool) error
 }
 
 // DownloadModelFiles downloads model files from S3.
 func DownloadModelFiles(ctx context.Context, s3Client s3Client, srcS3Path string, destDir string) error {
+	keys, err := listFiles(ctx, s3Client, srcS3Path)
```

Comment:
Currently some users don't have the `ListBucket` permission, so this change can break it. Can we gracefully handle the case?